worlds/rogue_legacy/__init__.py

```diff
@@ -1,176 +1,270 @@
+                        quantity = 1
+                if name == "Dragons":
+                    if "Dragon" not in self.setting("available_classes"):
+                        continue
+                if name == "Traitors":
+                    if "Traitor" not in self.setting("available_classes"):
+                        continue
+
+            # Skills
+            if name == "Health Up":
+                quantity = self.setting("health_pool")
+            elif name == "Mana Up":
+                quantity = self.setting("mana_pool")
+            elif name == "Attack Up":
+                quantity = self.setting("attack_pool")
+            elif name == "Magic Damage Up":
+                quantity = self.setting("magic_damage_pool")
+            elif name == "Armor Up":
+                quantity = self.setting("armor_pool")
+            elif name == "Equip Up":
+                quantity = self.setting("equip_pool")
+            elif name == "Crit Chance Up":
+                quantity = self.setting("crit_chance_pool")
+            elif name == "Crit Damage Up":
+                quantity = self.setting("crit_damage_pool")
+
+            # Ignore filler, it will be added in a later stage.
+            if data.category == "Filler":
+                continue
+
+            self.item_pool += [self.create_item(name) for _ in range(0, quantity)]
+
+        # Fill any empty locations with filler items.
+        while len(self.item_pool) + len(self.prefill_items) < total_locations:
+            self.item_pool.append(self.create_item(self.get_filler_item_name()))
+
+        self.world.itempool += self.item_pool
+
+    def pre_fill(self) -> None:
+        reachable = [loc for loc in self.world.get_reachable_locations(player=self.player) if not loc.item]
+        self.world.random.shuffle(reachable)
+        items = self.prefill_items.copy()
+        for item in items:
+            reachable.pop().place_locked_item(item)
+
+    def get_pre_fill_items(self) -> List[RLItem]:
+        return self.prefill_items
+
+    def get_filler_item_name(self) -> str:
+        fillers = get_items_by_category("Filler")
+        weights = [data.weight for data in fillers.values()]
+        return self.world.random.choices([filler for filler in fillers.keys()], weights, k=1)[0]
+
+    def create_item(self, name: str) -> RLItem:
         data = item_table[name]
-        return LegacyItem(name, ItemClassification.progression if data.progression else ItemClassification.filler, data.code, self.player)
+        return RLItem(name, data.classification, data.code, self.player)
+
+    def create_event(self, name: str) -> RLItem:
+        data = event_item_table[name]
+        return RLItem(name, data.classification, data.code, self.player)
 
     def set_rules(self):
         set_rules(self.world, self.player)
+
+    def create_regions(self):
+        create_regions(self.world, self.player)
+        self.__place_events()
+
+    def __place_events(self):
```

Comment:
any reason why this is `__place_events` and not `_place_events` ? Python does special handling for `__`, which makes single `_` equivalent to "protected", double `__` equivalent to private.

Reply:
Because it's supposed to be only used by this class, hence private.

Reply:
I don't see any world where something should inherit from this class.

Reply:
Oh, maybe let me rephrase. Python will rename the function from `__place_events` to `_RLWorld__place_events` to ensure it being "private". We normally only do that in python if we explicitly want something to be private for a reason. So if I see `__` I'm wondering what the reason for that is, and not the other way around.

Reply:
I understand the implementation details, I'm talking methodology. In my experience in other projects and languages, you typically want an explicit reason for making something more accessible, not less. You want to reduce the amount of API that isn't needed and make it easier if you need to refactor later.

Protected would should be fine, but I don't see a reason for it to be protected vs private if `RLWorld` isn't to be inherited from.

Might be different in philosophy in how we determine access modifiers.